src/snapshots/new_nu_parser__test__node_output@reparse.nu.snap

```diff
@@ -10,10 +10,11 @@ input_file: tests/reparse.nu
 4: Name (12 to 18)
 5: List([NodeId(2), NodeId(3), NodeId(4)]) (8 to 18)
 6: Name (22 to 26)
-7: Variable (27 to 29)
-8: Block(BlockId(0)) (22 to 30)
-9: Closure { params: None, block: NodeId(8) } (20 to 31)
-10: Name (33 to 36)
-11: Int (37 to 38)
-12: Block(BlockId(1)) (0 to 38)
+7: Garbage (30 to 31)
+8: Name (33 to 36)
+9: Int (37 to 38)
+10: Block(BlockId(0)) (22 to 38)
+11: Garbage (38 to 38)
```

Comment:
Why do we now have garbage here?

It seems to alter the parse drastically on failing to resolve the name?

Reply:
I think it's because we don't parse defs yet, so parsing the signature doesn't add a new variable which causes the `$x` inside the block to fail.

Reply:
In my view of the new parser `$bla` should still get marked as a `Variable` even if the symbol has not been resolved (yet). The grammar/syntax doesn't change or shouldn't change context dependently for `$` prefixed variables. For syntax highlighting or other LSP operations unbound symbols should still be easily queryable. 
Also this could quickly become unpredictable if we fail an expression because of that.

Reply:
OK, yeah, you're right. I updated it.